dashboard/app/models/user.rb

```diff
@@ -1761,6 +1761,9 @@ def clear_user_and_mark_purged
 
     authentication_options.with_deleted.each(&:really_destroy!)
 
+    districts.clear
+    self.district_as_contact = nil
```

Comment:
nit: `self.` is implied here, so this line can be: `district_as_contact = nil` (same with the lines below, although those aren't related to this PR)

Reply:
I'm not sure that's true.  My tests fail if I remove `self.` here, and `district_as_contact` is not cleared.  Rubymine's linter hints that without `self.` this gets interpreted as a local variable `district_as_contact` initialized to `nil`.  That interpretation lines up with [this StackOverflow answer](https://stackoverflow.com/questions/5068304/in-ruby-when-should-you-use-self-in-your-classes) and [the end of this blog post](https://www.jimmycuadra.com/posts/self-in-ruby/):

> Another time when self is needed for disambiguation is when assigning a value to one of the object's attributes.